app/views/file_attachments/edit/_official_document.html.erb

```diff
@@ -0,0 +1,30 @@
+<% official_document_type = params.dig(:file_attachment, :official_document_type) ||
+                            if attachment.unofficial? then "unofficial"
+                            elsif attachment.paper_number.present? then attachment.official_document_type
+                            elsif attachment.act_paper? then "unnumbered_act_paper"
+                            elsif attachment.command_paper? then "unnumbered_command_paper"
+                            end %>
+
+<% official_document_items = t("file_attachments.edit.official_document.options").map { |key, locales|
```

Comment:
I think it's a bit dubious to use the translation object as something to traverse on, I wouldn't expect them to be used for any logic or to allow any configuration - though I'm expecting you're mostly using it as a short cut. I'd suggest that you map through an array of the options and then look up the translation keys - which also feels a more realistic usage of the I18n monitoring.

Also, minor, but convention would dictate a do end block rather than braces.

Reply:
Updated 👍